Use uv_ssize_t type instead of ssize_t

ruslo opened the pull request:
Type ssize_t is not a part of the C++ standard and may not be available on some
platforms like Windows. Libuv currently declare ssize_t for such platforms in
public headers. This may lead to conflicts if other project try to declare
ssize_t too. Renaming ssize_t to uv_ssize_t fix this issue and conform code
style rule about [defining new types][1]:

```
All the symbols and types that libuv makes available publicly should be
prefixed with 'uv_'.
```

[1]: https://github.com/libuv/libuv/blob/v1.14.0/CONTRIBUTING.md#code

ruslo commented:
It would be nice to have same fix for stable `1.x` branch but I guess it's not acceptable because it will break API on Windows (?)

mgaf commented:
good

saghul commented:
Hi @ruslo! I see this is your first contribution, welcome aboard!

I like what I see, so I'm +1 on having it. Now, as you suspected, this change is not backwards compatible, so we can land it on master only. I see you already targetted your PR correctly, so that's great :-)

When I saw your patch it reminded me of one I had worked on for some time but forgot to merge: #1429. All include files are now in include/uv and thus don't need the uv- prefix in the filename. Can you please rebase your patch? Also, make sure the commit message follows the guidelines outlined in CONTRIBUTING.md, thanks a lot!

ruslo commented:
> Can you please rebase your patch?

Done

bwijen commented on docs/code/dns/main.c:
```diff
@@ -10,7 +10,7 @@ void alloc_buffer(uv_handle_t *handle, size_t suggested_size, uv_buf_t *buf) {
 }
 
-void on_read(uv_stream_t *client, ssize_t nread, const uv_buf_t *buf) {
+void on_read(uv_stream_t *client, uv_ssize_t nread, const uv_buf_t *buf) {
```
What about:
size_t nread
if(nread!=0) -> uv_buf_t is valid
else -> check uv_stream_t for errors.

ruslo replied:
This patch is about using `uv_ssize_t` instead of `uv_ssize_t`. I think discussion of logic is out of scope of this change.

saghul replied:
Agreed. @bwijen can you please create a new issue / PR with what you spotted? Thanks!

bwijen replied:
Agreed: #1548

ruslo commented:
Ping

cjihrig commented:
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/980/

ruslo commented:
> There are significant failures

I have fixed Windows failure, updated. Other failures are not related to this patch. E.g. https://ci.nodejs.org/job/libuv-test-commit-linux-cmake/57/nodes=centos7-64/console failed because there is no CMakeLists.txt in branch `master`, https://ci.nodejs.org/job/libuv-test-commit-linux/1023/nodes=centos6-64-gcc44/console builds fine but test failed because of timeout.
